Cyren ThreatInDepth Base Content Pack v1.2.0

meilke opened the pull request:
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

N/A

## Description

* indicator statistics dashboard
* moved custom indicator field mappings from Python integration code to dedicated indicator mapper
* wording improvements in documentation

## Screenshots

![image](https://user-images.githubusercontent.com/2668980/105057855-ee436600-5a75-11eb-8dd2-bdaefc7e4684.png)

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

meilke commented:
@barchen1 what are the next steps here?

barchen1 commented:
@meilke So first I will need to review your PR snd give you some comments and then you will need to fix them and that's it :)
I understand from @abaumgarten that already worked with her so you are familiar with our process.
Let me know if you have any questions

barchen1 reviewed (changes requested):
Thanks a lot for your contribution! You did a great job!
Here are some notes about your PR:
1. Please make sure that you don't have any secrets in your `CyrenThreatInDepth_test.py` file. You can run `demisto-sdk secrets`.
2. Can you please explain the changes you made to the `get_fields` function and why you changed it to `get_tags`?

Let me know if you have any questions.

meilke commented:
@barchen1, regarding removing `get_fields`: As you can see in the pull request we moved most of our custom field mapping from the integration Python code to an indicator mapper object. So I removed `get_fields` from the feed item class (only the diff shows it as a replacement with `get_tags`). Most of the fields can be retrieved from our raw JSON, but some fields need "preparation" (hence the source tag and the tags raw JSON fields) - it would have been to complicated to do it in the mapper only. This is also because we actually produce two different raw JSON objects (because of the relationships) and then you can prevent mapping transformer/filter magic.

Here is a little visualization:

![image](https://user-images.githubusercontent.com/2668980/105686044-13f9c080-5ef7-11eb-99ed-a9bf832b7263.png)

meilke commented on Packs/FeedCyrenThreatInDepth/Integrations/CyrenThreatInDepth/CyrenThreatInDepth.py:
```diff
@@ -93,8 +93,11 @@ def to_indicator_objects(self) -> List[Dict]:
             fields["cyrenfeedrelationships"] = relationship_indicators
 
+        raw_json = self.entry.copy()
+        raw_json["source_tag"] = FeedSource.PRIMARY
+        raw_json["tags"] = self.get_tags()
```
Because of field differences between primary and related indicators and because we want to include multiple values from multiple fields into the final `tags` field, we decided to go with these intermediate fields. They help and prepare data for the mapper.

meilke commented on Packs/FeedCyrenThreatInDepth/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,32 @@
+##### Cyren Threat InDepth Threat Intelligence Feed
+
+- moved custom field mapping functionality from the integration's source code to
+  a customizable indicator mapper
```
done

meilke commented on Packs/FeedCyrenThreatInDepth/Integrations/CyrenThreatInDepth/CyrenThreatInDepth.py:
```diff
@@ -138,20 +141,14 @@ def get_score(self) -> int:
-                      cyrendetectioncategories=self.categories,
-                      cyrendetectionmethods=detection_methods,
-                      cyrensourcetags=[FeedSource.PRIMARY])
+        fields = dict(updateddate=timestamp,
+                      indicatoridentification=self.payload.get("identifier"))
```
these are fields that I cannot set in the mapper UI, they are simply not available

barchen1 reviewed (changes requested):
Thanks for your fix and for your answers to my questions
1. Please update the `1_2_0.md` file.
2. Regarding the `get_fields` change, we will talk about it in the demo later today.

meilke commented on Packs/FeedCyrenThreatInDepth/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,28 @@
+
+##### New: Cyren Threat InDepth Dashboard
+
+- A new dashboard provides general information about the data pulled from Cyren Threat InDepth. It can be used directly as new dashboard or individual widgets can be added to existing dashboards (Available from Cortex XSOAR 6.0.0).
```
done

meilke commented on Packs/FeedCyrenThreatInDepth/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,28 @@
+
+##### New: Cyren Threat InDepth Indicator Mapper
+
+- The new Cyren Threat InDepth Indicator Mapper provides flexibility to gather the data the way they are needed it in the customer's process (Available from Cortex XSOAR 6.0.0).
```
done

meilke commented on Packs/FeedCyrenThreatInDepth/ReleaseNotes/1_2_0.md:
```diff
@@ -0,0 +1,28 @@
+- moved custom field mapping functionality from the integration's source code to a customizable indicator mapper
+  - new instances of the integration will have the **Cyren Threat InDepth Indicator Mapper** automatically included
+  - **for customers with existing instances**: make sure to use either the **Cyren Threat InDepth Indicator Mapper** or derive from it to adapt it to your needs
+
```
done
